rfc/20241023-rfc-tracking-issues.md

```diff
@@ -0,0 +1,209 @@
+
+One exception to the frozen state: if a new RFC is proposed that materially changes a decision made and recorded in an earlier accepted RFC, the PR for the new RFC should add a note in each relevant part of the older RFC that mentions that the decision has been changed and links to the new RFC, so that it's clear to a future reader of the old RFC that part of it has been obsoleted retroactively. Consider using [GitHub Markdown's "alert" extensions](https://github.blog/changelog/2023-12-14-new-markdown-extension-alerts-provide-distinctive-styling-for-significant-content/) to clearly separate the later-added content from the original frozen content, and add the new markers in the same PR that introduces the new RFC so that the changes are more strongly connected in the history.
+
+The "chains" of linked RFCs that might be caused by the frozen state could potentially become quite long over time for product areas that experience frequent change, but thankfully older RFCs tend to become less and less relevant over time. Any significant feature of the product or its codebases should also be documented in some separate location in a "how it currently is" shape, rather than a "how we changed it over time" shape, so that it isn't necessary to piece together information on currently-available features by trawling through the RFC history unless the historical change decisions _themselves_ are the point of interest.
+
+Some existing locations for longer-lived supporting documentation include:
+
+- [The OpenTofu docs](https://opentofu.org/docs/), maintained in the `website` directory in the root of the main OpenTofu repository, for end-user oriented documentation.
+- The `docs` directory in the root of the main OpenTofu repository, for developer-oriented descriptions of system architecture and the design of non-trivial features.
+- The `CONTRIBUTING.md` file and the RFC `README.md` file in the OpenTofu repository, for contributor oriented documentation about the development process rather than the code itself.
+- The documentation comments directly within our Go packages, attached to individual symbols and whole packages, for developer-oriented descriptions of fine-grain details specific the implementation of certain features.
+
+The tracking issue for an RFC should include work items for documation artifacts that the implementation team intends to create or update before the project is concluded, to reduce the risk of this step being forgotten. It is unlikely that any change large enough to justify an RFC does not also justify either an end-user-oriented or developer-oriented documentation change.
```

Comment:
I was _alluding_ to this sort of work in various places where I broadly referred to "non-coding tasks" rather than specifically "writing documentation".

I think I'd like to avoid getting into the weeds of what _exactly_ the "non-coding tasks" are, who is responsible for doing them, and how we track the need to do them in _this specific RFC_ just because it's already quite long, but I do fully agree that this is an important question to answer: I'd just prefer to get there iteratively as we learn from running this process in real life, rather than trying to get everything nailed down all in one step.

In particular this set of things seems to overlap with what I wrote under "Future Considerations" about whether our release planning model is like "it's not done until all of these features are complete" or like "choose an approx. freeze date, and include anything that was completed in time".

An key part of that tradeoff, I think, is whether we focus on "release blog posts" or "per-feature blog posts" (and other media, like video). If we decide that each feature gets its own separate blog post, etc then it could be reasonable for the RFC tracking issue to include the tasks of creating those, but if we treat it more like a "release rollup" then that work would presumably sometimes happen quite some time after a particular feature is complete, because not all features will necessarily complete development at the same point in a development period.

Again, I do completely agree that this is important stuff to decide, but I'd prefer not to get too into the specifics in _this_ RFC because our process for PR/marketing seems considerably less settled and thus far more likely to change quickly than our process for writing new code and documenting it. If you feel strongly about including it then I can be flexible, though.